src/AzureClient/EntryPoint/IEntryPointGenerator.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Quantum.IQSharp.AzureClient
+{
+    /// <summary>
+    /// This service is capable of generating entry points for
+    /// job submission to Azure Quantum.
+    /// </summary>
+    public interface IEntryPointGenerator
```

Comment:
If I make these interfaces internal, I have to also make the `AzureClient` constructor internal which takes this as a parameter, and this causes the service instantiation in our dependency injection framework to fail (since that happens from a non-IQ# assembly and requires a public constructor). So, long story short, I think they have to be public. Good question though, I wasn't fully aware of how that worked until I just looked at it.